github/pkg/adapter/adapter.go

```diff
@@ -47,13 +46,20 @@ type Adapter struct {
 func New(sinkURI, ownerRepo string) (*Adapter, error) {
 	a := new(Adapter)
 	var err error
-	a.client, err = kncloudevents.NewDefaultClient(sinkURI)
+	p, err := cloudevents.NewHTTP(cloudevents.WithTarget(sinkURI))
+	if err != nil {
+		log.Fatalf("failed to create http protocol: %s", err.Error())
+	}
+	a.client, err = cloudevents.NewClient(p, cloudevents.WithUUIDs(), cloudevents.WithTimeNow())
+	if err != nil {
+		log.Fatalf("failed to create client: %s", err.Error())
+	}
```

Comment:
We have a mix of fatal and returned errors in this function.
Since the function signature includes an error return value, we should always return the error, and let the caller fail however it wants to fail.